src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/AdditionalConditionWithConfig.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.AdditionalCondition
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.policy.IAdditionalCondition
+
+data class AdditionalConditionWithConfig(
+  val additionalCondition: AdditionalCondition,
+  val config: IAdditionalCondition,
+)
+
+fun mapConditionsToConfig(licenceConditions: List<AdditionalCondition>, policy: Set<IAdditionalCondition>): List<AdditionalConditionWithConfig> {
+  return licenceConditions.map {
+    val policyCondition = policy.find { pc -> it.conditionCode == pc.code }!!
+    AdditionalConditionWithConfig(it, policyCondition)
+  }
+}
+
+fun checkConditionsReadyToSubmit(licenceConditions: List<AdditionalCondition>, policy: Set<IAdditionalCondition>): Map<String, Boolean> {
+  val conditionsWithConfig = mapConditionsToConfig(licenceConditions, policy)
+  return conditionsWithConfig.associate {
+    val enteredFields = it.additionalCondition.additionalConditionData.map { data -> data.dataField }
+    val inputEntered =
+      if (!it.config.requiresInput) {
+        true
+      } else {
+        it.config.getConditionInputs()!!.map { input -> input.name }.any { name -> enteredFields.contains(name) }
```

Comment:
think maybe a a short comment saying that this relies on the assumption that any input provided means its ready to submit. Could also call `inputEntered`, `readyToSubmit` ? 

(that variable makes more sense for the situation when a condition doesn't require input too)